cmd/snap/cmd_run.go

```diff
@@ -333,7 +337,38 @@ func createUserDataDirs(info *snap.Info) error {
 		}
 	}
 
-	return createOrUpdateUserDataSymlink(info, usr)
+	if err := createOrUpdateUserDataSymlink(info, usr); err != nil {
+		return err
+	}
+
+	return maybeRestoreSecurityContext(filepath.Join(usr.HomeDir, dirs.UserHomeSnapDir))
+}
+
+// maybeRestoreSecurityContext attempts to restore security context of ~/snap on
+// systems where it's applicable
+func maybeRestoreSecurityContext(aPath string) error {
```

Comment:
the signature and the comment don't match, I think the comment makes sense tough, this helper would fit better with the rest if it had a signature like:

maybeRestoreUserDataSecurityContext(usr *user.User) , instead of taking a precoumpted directory